### DISCUSSION TITLE
Add loading prop to Checkbox

### DISCUSSION THREAD

**Discussion by juliusn** [mantinedev/mantine#6818] [Feature requests]
The `<Checkbox>` component could use similar `loading` prop as in `<Button>` for cases where the value is controlled and you need to complete a request before setting the value. Using the `disabled` prop is not as informative as, e.g. a spinning animation. There may be other input components for which the same feature could be added as well. 

```tsx
<Checkbox
  checked={checked}
  loading={loading}
  onChange={async (event) => {
    setLoading(true);
    // asynchronous code
    setLoading(false);
  }}
/>
```

**Comment by MariaBanaszkiewicz** [comment]
Hi! I also think this prop would be usefull. I would love to implement this feature! :)
